File: solutions/go/wordy/1/wordy.go
// Shree DR.MDD
package wordy

import (
	"regexp"
	"strconv"
	"strings"
)

var operations = map[string]func(int, int) int{
	"What is":       func(_, val int) int { return val },
	"plus":          func(a, b int) int { return a + b },
	"minus":         func(a, b int) int { return a - b },
	"multiplied by": func(a, b int) int { return a * b },
	"divided by":    func(a, b int) int { return a / b },
}

var validPattern = regexp.MustCompile(`^What is (-?\d+)( (plus|minus|multiplied by|divided by) (-?\d+))*\?$`)
var tokenPattern = regexp.MustCompile(`What is|plus|minus|multiplied by|divided by|-?\d+`)

func Answer(question string) (int, bool) {
	if !validPattern.MatchString(question) {
		return 0, false
	}

	tokens := tokenPattern.FindAllString(question, -1)
	var result int

	for i := 0; i < len(tokens)-1; i += 2 {
		opKey := strings.TrimSpace(tokens[i])
		val, err := strconv.Atoi(strings.TrimSpace(tokens[i+1]))
		if err != nil {
			return 0, false
		}
		opFunc, ok := operations[opKey]
		if !ok {
			return 0, false
		}
		result = opFunc(result, val)
	}

	return result, true
}

File: solutions/php/wordy/1/Wordy.php
<?php
// For my Shree DR.MDD

declare(strict_types=1);

function calculate(string $question): int
{
    if(strlen($question) < strlen("What is _ ?")) {
        throw new Exception();
    }
    if(substr($question, 0, 8) !== 'What is ') {
        throw new InvalidArgumentException();
    }
    
    $expression = substr($question, 8, strlen($question) - 9);
    $tokens = explode(' ', $expression);
    return evaluate((int)$tokens[0], array_slice($tokens, 1, count($tokens) - 1));
}

function evaluate(int $acc, array $tokens): int {
    print_r($tokens);
    if(count($tokens) === 0) {
        return $acc;
    }
    
    switch($tokens[0]) {
        case 'plus':
            return evaluate($acc + $tokens[1], array_slice($tokens, 2, count($tokens) - 2));
        case 'minus':
            return evaluate($acc - $tokens[1], array_slice($tokens, 2, count($tokens) - 2));
        case 'multiplied':
            return evaluate($acc * $tokens[2], array_slice($tokens, 3, count($tokens) - 3));
        case 'divided':
            return evaluate($acc / $tokens[2], array_slice($tokens, 3, count($tokens) - 3));
    }
    throw new InvalidArgumentException();
}
